dotnet/src/SemanticKernel/Kernel.cs

```diff
@@ -250,11 +254,19 @@ public async Task<SKContext> RunAsync(ContextVariables variables, CancellationTo
     }
 
     /// <inheritdoc/>
-    public SKContext CreateNewContext()
+    public SKContext CreateNewContext(ContextVariables? variables = null, IReadOnlyFunctionCollection? functions = null)
     {
-        return new SKContext(
-            this,
-            functions: this._functionCollection);
+#pragma warning disable CA2000 // Dispose objects before losing scope
+        var kernelContext = new KernelExecutionContext(
+            functions ?? new FunctionCollection(this._functionCollection),
+            this._aiServiceProvider,
+            this.PromptTemplateEngine,
+            this.Memory,
+            this.HttpHandlerFactory,
+            this.LoggerFactory);
```

Comment:
Can this be simplified to `var executionContext = new KernelExecutionContext(this, functions);`